internal/services/authorization/role_assignment_resource.go

```diff
@@ -284,15 +291,24 @@ func resourceArmRoleAssignmentRead(d *pluginsdk.ResourceData, meta interface{})
 		d.Set("condition_version", props.ConditionVersion)
 
 		// allows for import when role name is used (also if the role name changes a plan will show a diff)
-		if roleId := props.RoleDefinitionID; roleId != nil {
-			roleResp, err := roleDefinitionsClient.GetByID(ctx, *roleId)
+		if roleDefResourceId := props.RoleDefinitionID; roleDefResourceId != nil {
+			// The role definition id returned does not contain scope when the scope is some special case.
+			// Such as management group. So we might need to add scope here.
```

Comment:
Can we be more specific here that the modification is needed to workaround the issue: https://github.com/hashicorp/pandora/issues/3257 for `roledefinitions.ParseScopedRoleDefinitionID` for the roles assigned at the tenant level (both for tenant scope and mgmt group scope)?